zap/src/main/java/org/zaproxy/zap/spider/SpiderTask.java

```diff
@@ -234,22 +234,29 @@ private void runImpl() {
         parent.checkPauseAndWait();
 
         // Check the parse filters to see if the resource should be skipped from parsing
+        FilterResult filterResult = FilterResult.NOT_FILTERED;
         for (ParseFilter filter : parent.getController().getParseFilters()) {
-            FilterResult filterResult = filter.filtered(msg);
-            if (filterResult.isFiltered()) {
-                if (log.isDebugEnabled()) {
-                    log.debug(
-                            "Resource ["
-                                    + msg.getRequestHeader().getURI()
-                                    + "] fetched, but will not be parsed due to a ParseFilter rule: "
-                                    + filterResult.getReason());
-                }
-
-                parent.notifyListenersSpiderTaskResult(
-                        new SpiderTaskResult(msg, filterResult.getReason()));
-                return;
+            filterResult = filter.filtered(msg);
+            if (filterResult.isFiltered() || filterResult == FilterResult.WANTED) {
```

Comment:
This could skip some filters when `WANTED`, the suggestion was to call all of them to allow custom restrictions (regardless of the order they were added).